docs/resources/tozny_pam_jira_plugin.md

```diff
@@ -0,0 +1,139 @@
+for a group that allows a Jira board to create Access Requests and manage their approval/denial.
+
+## Example Usage
+
+```hcl
+###
+# Prerequisites & Setup
+#
+# A PAM Jira Plugin requires an MPC-enabled realm. This section sets up the runtime & tozny provider
+# and creates a realm we can work with.
+###
+
+# Configure the Terraform runtime
+terraform {
+  required_version = ">= 0.14"
+  required_providers {
+    tozny = {
+      # Pull signed provider binaries from the Terraform hosted registry namespace
+      # for Tozny registry.terraform.io/tozny
+      source = "tozny/tozny"
+      # Pin Tozny provider version
+      version = ">=0.14.0"
+    }
+  }
+}
+
+# Include the Tozny Terraform provider
+provider "tozny" {
+  api_endpoint     = "http://platform.local.tozny.com:8000"
+  account_username = "test+${random_string.account_username_salt.result}@tozny.com"
+}
+
+# Generate a random string for use in creating
+# accounts across environments or executions conflict free
+resource "random_string" "account_username_salt" {
+  length  = 8
+  special = false
+}
+
+# Generate a random string for use in creating
+# realms across environments or executions conflict free
+resource "random_string" "random_realm_name" {
+  length  = 8
+  special = false
+}
+
+# Local variables for defining where to store and find local Tozny client credentials
+locals {
+  tozny_client_credentials_filepath = "./tozny_client_credentials.json"
+}
+
+# A resource for provisioning a Tozny account using Terraform generated
+# credentials that are saved to user specified filepath for reuse upon success.
+resource "tozny_account" "autogenerated_tozny_account" {
+  autogenerate_account_credentials = true
+  persist_credentials_to           = "file"
+  client_credentials_save_filepath = local.tozny_client_credentials_filepath
+}
+
+# A resource for provisioning a TozID Realm
+# using local file based credentials for a Tozny Client with permissions to manage Realms.
+resource "tozny_realm" "my_organizations_realm" {
+  # Block on and use client credentials generated from the provisioned account
+  depends_on = [
+    tozny_account.autogenerated_tozny_account,
+  ]
+  client_credentials_filepath = local.tozny_client_credentials_filepath
+  realm_name                  = random_string.random_realm_name.result
+  sovereign_name              = "Administrator"
+  mpc_enabled                 = true # <-- important! we need MPC for our realm.
+}
+
+###
+# Managing your PAM Jira Plugin
+#
+# Once we have a realm, we can create a PAM Jira Plugin. This is the entity that allows
+# communication between TozID and Jira.
+###
+
+# Never commit your Jira API Key to source control.
+# To avoid this we create three variables pulled from the environment.
+# The names of the env variables should be prefixed with "TF_VAR_" when set in the shell.
+# If not set, terraform will prompt you for the values.
+
+# The Jira Host URL
+# export TF_VAR_JIRA_HOST_URL="tozid.atlassian.net"
```

Comment:
this `TF_VAR` is a new concept for our provider, so it needs rationale, and if the rationale is valid we should probably update the other use of environment variables(in the account resource for username and password) to also use this prefix

Reply:
it's just a suggestion. they're just terraform variables. i am happy to exclude these comments they just seemed helpful. if the env variables aren't found, terraform prompts you for the values

Reply:
After reading https://www.terraform.io/docs/language/values/variables.html#environment-variables and your comment more I get it. I was thinking that our tozny terraform resource for the jira plugin was manually looking for these variables, and so having it look for things like `TF_VAR_JIRA_HOST_URL` when currently we only look for `ACCOUNT_USERNAME` (not `TF_ACCOUNT_USERNAME`) would've been a new usage pattern.

Reply:
gotcha! yeah this is just some vanilla ol' terraform for setting a var automagically from the env